src/pages/storage/forms/StoragePoolClusteredSourceSelector.tsx

```diff
@@ -0,0 +1,49 @@
+import { Label } from "@canonical/react-components";
+import ClusterSpecificInput from "components/forms/ClusterSpecificInput";
+import { FormikProps } from "formik";
+import { FC } from "react";
+import { ensureEditMode } from "util/instanceEdit";
+import { StoragePoolFormValues } from "./StoragePoolForm";
+import { useClusterMembers } from "context/useClusterMembers";
+import { focusField } from "util/formFields";
+
+interface Props {
+  formik: FormikProps<StoragePoolFormValues>;
+}
+
+const StoragePoolClusteredSourceSelector: FC<Props> = ({ formik }) => {
+  const { data: clusterMembers = [] } = useClusterMembers();
+
+  const currentValues = Object.values(
+    formik.values.sourcePerClusterMember ?? {},
+  );
+
+  const memberNames: string[] = [];
+  clusterMembers.forEach((member) => memberNames.push(member.server_name));
+
+  return (
+    <>
+      <Label>Source</Label>
+      <ClusterSpecificInput
+        values={formik.values.sourcePerClusterMember}
+        id="sourcePerClusterMember"
+        isReadOnly={false}
+        onChange={(value) => {
+          void formik.setFieldValue("sourcePerClusterMember", value);
+        }}
+        toggleReadOnly={() => {
+          ensureEditMode(formik);
+          focusField("sourcePerClusterMember");
+        }}
+        memberNames={memberNames}
+        isDefaultSpecific={currentValues.some(
+          (item) => item !== currentValues[0],
+        )}
```

Comment:
I think this calculation is better moved into the `ClusterSpecificInput`. It has all the data to do the calculation locally.